hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/WritableStoreFactory.java

```diff
@@ -112,7 +112,9 @@ public <C> C getStore(@NonNull final Class<C> storeInterface) throws IllegalArgu
         final var entry = STORE_FACTORY.get(storeInterface);
         if (entry != null && serviceName.equals(entry.name())) {
             final var store = entry.factory().apply(states);
-            assert storeInterface.isInstance(store); // This needs to be ensured while stores are registered
+            if (!storeInterface.isInstance(store))
+                throw new IllegalArgumentException("No instance " + storeInterface
```

Comment:
`IllegalStateException` sounds more appropriate for `assert` statements than `IllegalArgumentException`

Reply:
Hmm I see.
I can update the diffs related to `storeInterface` and probably [this one](https://github.com/hashgraph/hedera-services/blob/fix-sonar-errors-in-hedera-app/hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java#L369) as well. Also for the `storeInterface` diffs I guess I should update the other `throw` that is a couple of lines below to be for `IllegalStateException`. wdyt?

Reply:
I don't have any specific preference for one exception over the other. Thinking about it today it seems less important, as long as one of the exceptions is thrown.

Reply:
I don't have specific preference either so I will leave it as it is for now.